spotify-controller.el

```diff
@@ -109,6 +112,28 @@ Apply SUFFIX to spotify-prefixed functions, applying ARGS."
     (unless (string= suffix "player-status")
       (spotify-player-status))))
 
+(defun spotify-update-metadata (metadata)
+  "Compose the playing status string to be displayed in the mode-line from METADATA."
+  (let* ((player-status spotify-player-status-format)
+         (duration-format "%m:%02s")
+         (json-object-type 'hash-table)
+         (json-key-type 'symbol)
+         (json (condition-case nil
+                   (json-read-from-string metadata)
+                 (error (spotify-update-player-status "")
+                        nil))))
+    (when json
+      (progn
+        (setq player-status (replace-regexp-in-string "%a" (truncate-string-to-width (gethash 'artist json) spotify-player-status-truncate-length 0 nil "...") player-status))
+        (setq player-status (replace-regexp-in-string "%t" (truncate-string-to-width (gethash 'name json) spotify-player-status-truncate-length 0 nil "...") player-status))
+        (setq player-status (replace-regexp-in-string "%n" (number-to-string (gethash 'track_number json)) player-status))
+        (setq player-status (replace-regexp-in-string "%l" (format-seconds duration-format (/ (gethash 'duration json) 1000)) player-status))
+        (setq player-status (replace-regexp-in-string "%s" (spotify-player-status-shuffling-indicator (gethash 'player_shuffling json)) player-status))
+        (setq player-status (replace-regexp-in-string "%r" (spotify-player-status-repeating-indicator (gethash 'player_repeating json)) player-status))
+        (setq player-status (replace-regexp-in-string "%p" (spotify-player-status-playing-indicator (gethash 'player_state json)) player-status))
+        (spotify-update-player-status player-status)
+        (setq spotify-player-metadata json)))))
+
```

Comment:
This function is mostly identical to [this one](https://github.com/danielfm/spotify.el/blob/e4d02a95a911edb5ebf6fa1d9a7db76be9f56b4f/spotify-controller.el#L138).

Why did you have to create a new function for this?

Reply:
Oof. I was merging in an old branch. Thanks for catching this. Lemme clean that up.

Reply:
@danielfm Updated!